code/modules/mining/equipment/mining_charges.dm

```diff
@@ -0,0 +1,215 @@
+		if(hacked)
+			message_admins("[ADMIN_LOOKUPFLW(user)] planted [name] on [target.name] at [ADMIN_COORDJMP(target)]")
+			log_game("planted [name] on [target.name] at [COORD(target)]", user)
+		installed = TRUE
+		target.overlays += image_overlay
+
+/obj/item/grenade/plastic/miningcharge/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/detonator))
+		return
+	var/obj/item/detonator/detonator = I
+	if(!(src in detonator.bombs) && !timer_off)
+		detonator.bombs += src
+		timer_off = TRUE
+		to_chat(user, "<span class='notice'>You synchronized [src] to a detonator.</span>")
+		playsound(src, 'sound/machines/twobeep.ogg', 50)
+		detonator.update_icon()
+	else
+		to_chat(user, "<span class='warning'>[src] was already synchronized to a existing detonator!</span>")
+
+/obj/item/grenade/plastic/miningcharge/proc/detonate()
+	addtimer(CALLBACK(src, PROC_REF(prime)), 3 SECONDS)
+
+/obj/item/grenade/plastic/miningcharge/prime()
+	if(hacked) //try not to blow your fingers off
+		explode()
+		return
+	var/turf/simulated/mineral/location = get_turf(target)
+	var/datum/effect_system/smoke_spread/S = new
+	S.set_up(smoke_amount, 0, location, null)
+	S.start()
+	//location.attempt_drill(null,TRUE,3) //Ru-paradise uses special hardness system - some rocks can be destroyed only after 2 or 3 hits. It stays here as the reminder that ss220 lavaland is better.
+	for(var/turf/simulated/mineral/rock in circlerangeturfs(location, boom_sizes[3]))
+		var/distance = get_dist_euclidian(location, rock)
+		if(distance <= boom_sizes[3])
+			rock.mineralAmt += 3 // if rock is going to get drilled, add bonus mineral amount
+			rock.gets_drilled()
+	for(var/mob/living/carbon/C in circlerange(location, boom_sizes[3]))
+		if(ishuman(C)) //working on everyone
+			var/distance = get_dist_euclidian(location, C)
+			C.flash_eyes()
+			C.Weaken((boom_sizes[2] - distance) * 1 SECONDS) //1 second for how close you are to center if you're in range
+			C.AdjustDeaf((boom_sizes[3] - distance) * 5 SECONDS) //guaranteed deafness
+			var/obj/item/organ/internal/ears/ears = C.get_int_organ(/obj/item/organ/internal/ears)
+			if(istype(ears) && C.check_ear_prot() < HEARING_PROTECTION_MINOR) //headsets should be enough to avoid taking damage
+				ears.receive_damage((boom_sizes[3] - distance) * 2) //something like that i guess. Mega charge makes 12 damage to ears if nearby
+			to_chat(C, "<span class='userwarning'>You are knocked down by the power of the mining charge!</span>")
+	qdel(src)
+
+/obj/item/grenade/plastic/miningcharge/proc/explode() //c4 code
+	var/turf/location
+	if(target)
+		if(!QDELETED(target))
+			if(isturf(target))
+				location = get_turf(target)
+			else
+				location = get_atom_on_turf(target)
+			target.overlays -= image_overlay
+	else
+		location = get_atom_on_turf(src)
+	if(location)
+		explosion(location, boom_sizes[1], boom_sizes[2], boom_sizes[3], cause = src)
+		location.ex_act(2, target)
+	if(ismob(target))
+		var/mob/M = target
+		M.gib()
```

Comment:
Is it even possible to attach these to people? I don't think our C4 supports that